scapy/contrib/isotp/isotp_soft_socket.py

```diff
@@ -773,10 +795,17 @@ def _recv_sf(self, data, ts):
             self.rx_state = ISOTP_IDLE
 
         length = data[0] & 0xf
+        if self.fd and len(data) > CAN_MAX_DLEN:
```

Comment:
I suggest to check if `length` from the previous calculation is 0 and `self.fd` is set. Maybe also ensure the frame has more than two bytes

```
if self.fd and length == 0 and len(data) >= 2:
   ...
```